internal/model/model.go

```diff
@@ -575,6 +579,12 @@ func (m *Model) ClusterConfig(deviceID protocol.DeviceID, cm protocol.ClusterCon
 	}
 	m.pmut.Unlock()
 
+	if m.upgradeTimer != nil && upgrade.CompareVersions(cm.ClientVersion, m.clientVersion) == upgrade.Newer {
```

Comment:
We should probably check the client name here as well, so we only do this for higher versions of syncthing, not for "Pulse v42" or whatever.